src/repo-query/pages/chat.tsx

```diff
@@ -0,0 +1,242 @@
+import { useEffect, useState } from "react";
+import { RepoQueryPages } from "../components/Dialog";
+import { REPO_QUERY_QUERY_ENDPOINT } from "../../constants";
+import ReactMarkdown from "react-markdown";
+import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
+import { darcula } from "react-syntax-highlighter/dist/esm/styles/prism";
+
+interface ChatMessage {
+    message: string;
+    isUser: boolean;
+}
+
+export const Chat = ({ ownerName, repoName, setCurrentPage }: { ownerName: string, repoName: string, setCurrentPage: (page: RepoQueryPages) => void }) => {
+    // chat ui with bubbles and input
+    const [chatHistory, setChatHistory] = useState<ChatMessage[]>([]);
+    const [sendEnabled, setSendEnabled] = useState(true);
+    const [botThinking, setBotThinking] = useState(false);
+    const [statusMessage, setStatusMessage] = useState("Thinking...🧠");
+
+    const processChunk = (chunk: string) => {
+        const chunkLines = chunk.split("\n");
+        const eventLine = chunkLines[0];
+        const dataLine = chunkLines[1];
+
+        const event = eventLine.split(": ")[1];
+        let data = { path: "" };
+
+        try {
+            data = JSON.parse(dataLine.split(": ")[1]);
+        } catch (e) {
+            // untermindated stirng
+            console.log(e, dataLine);
+        }
+
+
+        if (event === "SEARCH_CODEBASE") {
+            setStatusMessage("Searching the codebase for your query...🔍");
+        } else if (event === "SEARCH_FILE") {
+            setStatusMessage(`Searching ${data.path} for your query...🔍`);
+        } else if (event === "GENERATE_RESPONSE") {
```

Comment:
```suggestion
        } else if (event === "SEARCH_PATH") {
            setStatusMessage(`Looking for ${data.path} in the codebase...🔍`);
        } else if (event === "GENERATE_RESPONSE") {
```

Reply:
The Repo-query service will emit this event if the model responds with the path search function to look for a specific file. For user queries like: "What does the github.ts file do?"